examples/cc3200/xively_firmware_updates/main.c

```diff
@@ -0,0 +1,1249 @@
+//  LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE,
+//  DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY
+//  THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+//  (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+//  OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+//
+//*****************************************************************************
+
+
+/*****************************************************************************
+ *                              Includes
+ *****************************************************************************/
+
+/* Simplelink */
+#include "simplelink.h"
+#include "flc.h"
+
+/* Driverlib */
+#include "hw_gpio.h"
+#include "hw_ints.h"
+#include "hw_memmap.h"
+#include "hw_types.h"
+#include "interrupt.h"
+#include "pin.h"
+#include "prcm.h"
+#include "rom.h"
+#include "rom_map.h"
+#include "utils.h"
+#include "gpio.h"
+#include "stdio.h"
+
+/* Common interfaces */
+#include "button_if.h"
+#include "common.h"
+#include "gpio_if.h"
+#include "i2c_if.h"
+#include "pinmux.h"
+#include "tmp006drv.h"
+#include "uart_if.h"
+
+/* Xively Application */
+#include "time.h"
+#include "xi_bsp_rng.h"
+#include "xi_bsp_time.h"
+#include "xively.h"
+#include "xi_sft.h"
+#include "config_file.h"
+
+
+/*****************************************************************************
+ *                              Defines
+ *****************************************************************************/
+#define XIVELY_CRED_LEN_MAX ( 64 )
+#define WIFI_KEY_LEN_MAX ( 64 )
+#define XIVELY_CRED_LEN_MAX ( 64 )
+
+/* Path to load the configuration file from device's flash file system.
+ * This config file contains Xively connection credentials and WiFi SSID
+ * and password credentials */
+#define XIVELY_CFG_FILE "/etc/xively_cfg.txt"
+
+/* Some defaults for the optional connection address configuration values */
+#define XIVELY_DEFAULT_BROKER "broker.xively.com"
+#define XIVELY_DEFAULT_PORT ( 8883 )
+
+/* Xively Configuration File Key Names. These will be used to lookup key/value pairs
+ * for Xively credentials and WiFI credentials from a file stored in the device's
+ * flash file system */
+#define WIFI_SEC_TYPE_KEY "wifi_security_type"
+#define WIFI_SSID_KEY "wifi_ssid"
+#define WIFI_PASSWORD_KEY "wifi_password"
+#define XIVELY_HOST_KEY "hostname"
+#define XIVELY_PORT_KEY "port"
+#define XIVELY_ACNT_KEY "account_id"
+#define XIVELY_DEV_KEY "device_id"
+#define XIVELY_PWD_KEY "password"
+
+/* Used to detect if the user has configured their xively_config.txt file
+ * on the device before the execution of the application */
+#define XIVELY_DEFAULT_CONFIG_FIELD_VALUE "<PASTE_XIVELY_CREDENTIAL_HERE>"
+#define WIFI_DEFAULT_CONFIG_FIELD_VALUE "<ENTER_WIFI_CREDENTIAL_HERE>"
+
+/* String template for dynamically constructing the various Xively topics
+ * that are used for this application. */
+#define XIVELY_TOPIC_LEN ( 128 )
```

Comment:
is 128 enough? can't customer easily enter longer topic name? I mean there is no enough large number but isn't 256 more safe? just asking